backend/driver/driver.go

```diff
@@ -0,0 +1,211 @@
+	JobNamespace string
+	JobName      string
+	ConfigMap    string
+	OutputPath   string
+	Logger       logr.Logger
+	Args         []string
+}
+
+type Driver interface {
+	Run() error
+}
+
+type driverImpl struct {
+	client client.Client
+	job    lmevalservicev1beta1.EvalJob
+	Option *DriverOption
+}
+
+func NewDriver(opt *DriverOption) (Driver, error) {
+	if opt == nil {
+		return nil, nil
+	}
+
+	if opt.Context == nil {
+		return nil, fmt.Errorf("context is nil")
+	}
+
+	if opt.JobNamespace == "" || opt.JobName == "" {
+		return nil, fmt.Errorf("JobNamespace or JobName is empty")
+	}
+
+	return &driverImpl{Option: opt}, nil
+}
+
+// Run implements Driver.
+func (d *driverImpl) Run() error {
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	client, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	d.client = client
+	if err := d.getJob(); err != nil {
+		return err
+	}
+
+	if err := d.updateStatus(lmevalservicev1beta1.RunningJobState); err != nil {
+		return err
+	}
+
+	err = d.exec()
+
+	if err := d.getJob(); err != nil {
+		return err
+	}
+
+	return d.updateCompleteStatus(err)
+}
+
+func (d *driverImpl) getJob() error {
+	if err := d.client.Get(d.Option.Context,
+		types.NamespacedName{
+			Name:      d.Option.JobName,
+			Namespace: d.Option.JobNamespace,
+		},
+		&d.job); err != nil {
+
+		return err
+	}
+	return nil
+}
+
+func (d *driverImpl) exec() error {
+	// Run user program.
+	var args []string
+	if len(d.Option.Args) > 1 {
+		args = d.Option.Args[1:]
+	}
+
+	stdout, err := os.Create(filepath.Join(d.Option.OutputPath, "stdout.log"))
+	if err != nil {
+		return err
+	}
+	bout := bufio.NewWriter(stdout)
+
+	stderr, err := os.Create(filepath.Join(d.Option.OutputPath, "stderr.log"))
+	if err != nil {
+		return err
+	}
+	berr := bufio.NewWriter(stderr)
+
+	defer func() {
+		bout.Flush()
+		stdout.Sync()
+		stdout.Close()
+		berr.Flush()
+		stderr.Sync()
+		stderr.Close()
+	}()
+
+	fmt.Printf("args:%v\n", args)
+	executor := exec.Command(d.Option.Args[0], args...)
+	executor.Stdin = os.Stdin
+	executor.Stdout = bout
+	executor.Stderr = berr
+
+	if err := executor.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *driverImpl) updateStatus(state lmevalservicev1beta1.JobState) error {
+	if d.job.Status.State != state {
+		d.job.Status.State = state
+		if err := d.client.Status().Update(d.Option.Context, &d.job); err != nil {
+			d.Option.Logger.Error(err, "unable to update EvalJob.Status.State")
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *driverImpl) updateCompleteStatus(err error) error {
+	d.job.Status.State = lmevalservicev1beta1.CompleteJobState
+	d.job.Status.Reason = lmevalservicev1beta1.SucceedReason
+	if err != nil {
+		d.job.Status.Reason = lmevalservicev1beta1.FailedReason
+		d.job.Status.Message = err.Error()
+	} else {
+		// read the content of result*.json
+		pattern := filepath.Join(d.Option.OutputPath, "result*.json")
```

Comment:
Is this `"result*.json"` something that is static to `lm-eval`, or is it something a user might configure at some point with a custom `task`?

Reply:
that's static for `lm-eval`. but I will keep an eye on that to see if it's still true in the future release of lm-eval